initializr-generator/src/main/groovy/io/spring/initializr/generator/ProjectGenerator.groovy

```diff
@@ -199,6 +200,15 @@ class ProjectGenerator {
 		def model = [:]
 		def metadata = metadataProvider.get()
 
+		def useCustomParent = metadata.configuration.env.customParentPomGAV
+		model['useCustomParentPom'] = useCustomParent != null
+		if (useCustomParent) {
+			def gavParts = useCustomParent.split(':')
+			model['customParentPomGroup'] = gavParts[0]
+			model['customParentPomArtifact'] = gavParts[1]
+			model['customParentPomVersion'] = gavParts[2]
+			request.boms.put("spring-boot", new BillOfMaterials(groupId: "org.springframework.boot", artifactId: "spring-boot-dependencies", version: request.bootVersion))
+		}
```

Comment:
Adding the spring boot bom is something that should be configurable. Your own parent may inherit from that parent making the bom import unnecessary. I'd add a method to add  `spring-boot-dependencies` rather than bake that inside that method.